lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,21 @@
+module Semian
+  class AtomicInteger < SharedMemoryObject #:nodoc:
+    attr_accessor :value
+
+    def initialize(name, permissions)
+      @value=0
+    end
+
+    def increase_by(val)
```

Comment:
Maybe a different method name would make it clearer? The purpose is to increase the stored value by a set number. When the C implementation is added, keeping the entire mutating in one function is how an operation like `atomicinteger.value += 5` can maintain atomicity.